### PR TITLE
internal/providers/azure: add support for ssh keys

### PR DESCRIPTION
**Pull request by cgonyeo**
This PR adds support for fetching the user's ssh key in azure

Not ready for merge yet.

### CODE REVIEW

**Comment by lucab** [comment]
I have to say I'm not super happy about having exec-wrappers around openssl and ssh-keygen and feeding them (plaintext) network input. Is there any reason for not using go crypto&codec primitives from stdlib?

File: internal/providers/azure/azure.go [hunk 1]
```diff
@@ -230,3 +277,158 @@ func fetchSharedConfig(endpoint net.IP) (metadata, error) {
...
+	if err != nil {
+		return "", err
+	}
+	return pubkey, nil
+}
+
+// genCerts will generate a new 2048 bit rsa key and certificate for use in
+// certificate retrieval.
+func genCerts(keyFilePath, certFilePath string) error {
+	err := exec.ExecExpectZero("openssl", "req", "-x509", "-nodes", "-subj", "/CN=LinuxTransport", "-days", "32768", "-newkey", "rsa:2048", "-keyout", keyFilePath, "-out", certFilePath)
```
Comment (lucab) [inline internal/providers/azure/azure.go:363]: Are the common-name and expiration mandated somewhere? If so, please add a reference. If not, please use some meaningful cn and a shorter expiration (or leave a note why you are using such a large one, eg. early-boot time skews).

**Review by lucab — CHANGES_REQUESTED** [review]

File: internal/providers/azure/azure.go (hunk 1 above)
Comment (cgonyeo) [reply internal/providers/azure/azure.go:363]: I'm not sure if they're mandated, this command was directly copied from the waagent code: https://github.com/Azure/WALinuxAgent/blob/master/azurelinuxagent/common/utils/cryptutil.py#L30

I can play around with changing the CN and shortening the expiration, if it works with more reasonable values I'll change it to use them.

**Comment by cgonyeo** [comment]
There isn't a strong reason for using exec-wrappers other than this is how the waagent works, and it took me less work to implement it here the same way it works in waagent than to work through how to do this in pure go. I wouldn't be opposed to spending the time on switching over to the golang stdlib, but I can't say how much time it would take me to figure that out.

**Comment by crawford** [comment]
> it took me less work to implement it here the same way it works in waagent than to work through how to do this in pure go

I'm not interested in work being done unless it's being done correctly. Please investigate doing it properly.

**Pull request closed without merging**